File: src/3345.smallest-divisible-digit-product-i.3626/solution.java
class Solution {
    public int smallestNumber(int n, int t) {

        while (multi(n) % t != 0) {
            n++;
        }
        return n;
    }
    
    private int multi(int n){
        int result = 1;
        while (n != 0){
            result *= (n%10);
            n /= 10;
        }
        return result;
    }
}

File: src/3345.smallest-divisible-digit-product-i.3626/solution.ts
const multi = (n:number) => {
    let result = 1;
    while (n > 1) {
        const digit = n % 10;
        result *= digit;
        n = (n - digit) / 10;
    }
    return result;
};

function smallestNumber (n: number, t: number): number {
    while (multi(n) % t !== 0) {
        n++;
    }
    return n;
}
